pumpkin-inventory/src/lib.rs

```diff
@@ -52,3 +62,185 @@ impl WindowType {
         "WINDOW TITLE"
     }
 }
+// Container needs Sync + Send to be able to be in async Server
+pub trait Container: Sync + Send {
+    fn window_type(&self) -> &'static WindowType;
+
+    fn handle_item_change(
+        &mut self,
+        carried_item: &mut Option<ItemStack>,
+        slot: usize,
+        mouse_click: MouseClick,
+    ) -> Result<(), InventoryError> {
+        let mut all_slots = self.all_slots();
+        if slot > all_slots.len() {
+            Err(InventoryError::InvalidSlot)?
+        }
+        handle_item_change(carried_item, all_slots[slot], mouse_click);
+        Ok(())
+    }
+
+    fn all_slots(&mut self) -> Vec<&mut Option<ItemStack>>;
+
+    fn all_slots_ref(&self) -> Vec<Option<&ItemStack>>;
+
+    fn all_combinable_slots(&self) -> Vec<Option<&ItemStack>> {
+        self.all_slots_ref()
+    }
+
+    fn all_combinable_slots_mut(&mut self) -> Vec<&mut Option<ItemStack>> {
+        self.all_slots()
+    }
+
+    fn print_all_contents(&self) {
+        self.all_slots_ref()
+            .into_iter()
+            .enumerate()
+            .filter_map(|(slot, item)| item.map(|item| (slot, item)))
+            .for_each(|(slot, item)| {
+                dbg!(slot, item);
+            });
+    }
+
+    fn internal_pumpkin_id(&self) -> u64 {
+        0
+    }
```

Comment:
I still don't got for what that used for

Reply:
Placeholder for now. My thought process goes that instead of storing Containers on the chunks in the `World` struct, the blocks only store a unique ID, which can be used to access the inventory. So we don't need to block `World` and chunks when interacting with inventories.